### ISSUE TITLE
Does not build against nixos-22.11 revision d41eea2839de7de0f2cbe4aa90fde5db8afcd30a

### ISSUE THREAD

**Issue by kjeremy** [symphorien/nixseparatedebuginfod#6]
This builds for me against nixos-unstable but my coworker is having issues on nixos-22.11 revision d41eea2839de7de0f2cbe4aa90fde5db8afcd30a.

I'm seeing the following error

```console
    'git+file:///nix/store/yrxb3y71d0ksig0cg1zmfa09wrhlihjj-source/ipd-nix' (2023-03-09)
  → 'git+file:///nix/store/z0yi2qlh59mxs31jl6m481jg2pbkwacn-source/ipd-nix' (2023-03-09)
error: builder for '/nix/store/njhbqgbway0799960ckckbkxgamvmgvj-rust_nixseparatedebuginfod-0.1.0.drv' failed with exit code 1;
       last 10 log lines:
       > 112 | /     let Some(storepath) = get_store_path(exe.as_path()) else {
       > 113 | |         anyhow::bail!("executable {} for buildid {} is not a store path", exe.display(), buildid);
       > 114 | |     };
       >     | |______^
       >     |
       >     = note: see issue #87335 <[https://github.com/rust-lang/rust/issues/87335>](https://github.com/rust-lang/rust/issues/87335%3E) for more information
       >
       > error: aborting due to 3 previous errors
       >
       > For more information about this error, try `rustc --explain E0658`.
       For full logs, run 'nix log /nix/store/njhbqgbway0799960ckckbkxgamvmgvj-rust_nixseparatedebuginfod-0.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/sf7g3gf56fbgy8zwryggbifbvlkyd5mq-unit-nixseparatedebuginfod.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5m99izqdy9p3sg82ppkyapbk8c17s7g0-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hf91fh1i7mp7qrxph1pqjf64m8mvd00l-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gfp14yvpm2n0pacnn1hf48r3y49nqka5-nixos-system-lrogowski_host-22.11.20230515.d41eea2.drv' failed to build
```

**Comment by SuperSandro2000** [comment]
The important log is only shown when you run ``nix log /nix/store/njhbqgbway0799960ckckbkxgamvmgvj-rust_nixseparatedebuginfod-0.1.0.drv``.

**Comment by kjeremy** [comment]
```shell
nix log /nix/store/njhbqgbway0799960ckckbkxgamvmgvj-rust_nixseparatedebuginfod-0.1.0.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/njhbqgbway0799960ckckbkxgamvmgvj-rust_nixseparatedebuginfod-0.1.0.drv!*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/mvgqqvxw9qggccrgl8ix7lgflkjk8vhn-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
Running cd .
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building nixseparatedebuginfod (src/main.rs)
Running rustc --crate-name nixseparatedebuginfod src/main.rs --crate-type bin -C opt-level=3 -C codegen-units=1 --remap-path-prefix=/build=/ --extern anyhow=/nix/store/8h34w40g0aw34h8a9l0nqc910am48iip-rust_anyhow-1.0.68-lib/lib/libanyhow>
error[E0658]: `let...else` statements are unstable
   --> src/index.rs:262:9
    |
262 | /         let Some(path) = get_store_path(Path::new(path)) else {
263 | |             anyhow::bail!(
264 | |                 "read corrupted stuff from nix db: {}, concurrent write?",
265 | |                 path
266 | |             );
267 | |         };
    | |__________^
    |
    = note: see issue #87335 <[https://github.com/rust-lang/rust/issues/87335>](https://github.com/rust-lang/rust/issues/87335%3E) for more information

 

error[E0658]: `let...else` statements are unstable
   --> src/server.rs:109:5
    |
109 |     let Some(exe) = cache.get_executable(buildid).await.with_context(|| format!("getting executable of {} from cache", buildid))? else { return Ok(()) };
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #87335 <[https://github.com/rust-lang/rust/issues/87335>](https://github.com/rust-lang/rust/issues/87335%3E) for more information

 

error[E0658]: `let...else` statements are unstable
   --> src/server.rs:112:5
    |
112 | /     let Some(storepath) = get_store_path(exe.as_path()) else {
113 | |         anyhow::bail!("executable {} for buildid {} is not a store path", exe.display(), buildid);
114 | |     };
    | |______^
    |
    = note: see issue #87335 <[https://github.com/rust-lang/rust/issues/87335>](https://github.com/rust-lang/rust/issues/87335%3E) for more information

 

error: aborting due to 3 previous errors

 

For more information about this error, try `rustc --explain E0658`.
```

**Comment by kjeremy** [comment]
The rust version probably needs to be explicit.

**Comment by symphorien** [comment]
Fixed in 4b50a3d9e6278d533e3a81e3519d8b61842e3ee7
